src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -1,18 +1,19 @@
 (ns puppetlabs.trapperkeeper.authorization.acl
-  (:require [schema.core :as schema]
-            [clojure.string :as str]))
+  (:require [schema.core :as schema]))
 
 ;; Schemas
 
-(def auth-type (schema/enum :allow :deny))
+(def AuthType (schema/enum :allow :deny))
+(def EntryPattern (schema/conditional
+                   string? String
+                   sequential? [String]
+                   :else schema/Regex))
```

Comment:
Oh, this is the pattern for an individual `Entry`, right.  In that case, I think my comment doesn't apply.  Suppose the `sequentlal? [String]` is there because tk-auth parses a domain string ACE out into that for its run-time use.